integration-tests/src/test/java/com/myhome/controllers/UserControllerIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+package com.myhome.controllers;
+
+import com.myhome.MyHomeServiceApplication;
+import com.myhome.domain.User;
+import com.myhome.model.CreateUserRequest;
+import com.myhome.repositories.UserRepository;
+import io.restassured.http.ContentType;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.context.WebApplicationContext;
+
+import static io.restassured.module.mockmvc.RestAssuredMockMvc.given;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyOrNullString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = MyHomeServiceApplication.class,
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestInstance(TestInstance.Lifecycle.PER_METHOD)
+public class UserControllerIntegrationTest {
```

Comment:
Let's switch to `TestRestTemplate` and make this class similar to this:
```
@ExtendWith(SpringExtension.class)
@SpringBootTest(
    classes = MyHomeServiceApplication.class,
    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
)
@TestInstance(TestInstance.Lifecycle.PER_METHOD)
public class UserControllerIntegrationTest {

  private static final String TEST_NAME = "name";
  private static final String TEST_EMAIL = "email@mail.com";
  private static final String TEST_PASSWORD = "password";

  @LocalServerPort
  private int port;
  private TestRestTemplate restTemplate;

  @BeforeEach
  public void setUp() {
    restTemplate = new TestRestTemplate();
  }

  @Test
  public void shouldSignUpSuccessful() {
    // given
    final CreateUserRequest requestBody = new CreateUserRequest()
        .name(TEST_NAME)
        .email(TEST_EMAIL)
        .password(TEST_PASSWORD);
    final String requestUrl = buildUrl(port, REGISTRATION_PATH);
    final User expectedBody = new User()
        .withName(TEST_NAME)
        .withEmail(TEST_EMAIL);

    // when
    final ResponseEntity<User> responseEntity =
        restTemplate.postForEntity(requestUrl, requestBody, User.class);
    final User actualBody = responseEntity.getBody();

    // then
    assertThat(responseEntity.getStatusCode())
        .isEqualTo(HttpStatus.CREATED);
    assertThat(actualBody)
        .usingRecursiveComparison()
        .ignoringFields("userId")
        .isEqualTo(expectedBody);
  }
}
```

Reply:
Done.

I would still recommend exploring RestAssured in the future, because RestAssured's errors logs on failed tests are even more useful than regular assertion errors. Anyway, the switch to `TestRestTemplate` is complete.

Reply:
@rmgrimm for sure I will read more about it, thanks for that 👍